### PR TITLE
Feature: raid events

### PR DESCRIPTION
**Pull request by JayJay1989**
This will implement raid events:

 - raid_update
 - raid_update_v2
 - raid_go

This is only for outgoing raids

payload: https://gist.github.com/JayJay1989/2f3049ac14ea293fe7d866b2125beab9

This is undocumented in the twitch docs.

### CODE REVIEW

File: TwitchLib.PubSub/Events/OnRaidGoArgs.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    /// <inheritdoc />
+    /// <summary>
+    /// Object representing the arguments for a raid go event
+    /// </summary>
+    public class OnRaidGoArgs : EventArgs
+    {
+        /// <summary>
+        /// The channel ID the event came from
+        /// </summary>
+        public int ChannelId;
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Events/OnRaidGoArgs.cs]: Twitch wants third party devs to treat user/channel ids as strings.

File: TwitchLib.PubSub/Events/OnRaidGoArgs.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        /// <summary>
+        /// Property representing the id of the raid event
+        /// </summary>
+        public Guid Id;
+        /// <summary>
+        /// Property representing the target channel id
+        /// </summary>
+        public int TargetChannelId;
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Events/OnRaidGoArgs.cs]: Same here (string as a channel/user id)

File: TwitchLib.PubSub/Events/OnRaidUpdateArgs.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    /// <inheritdoc />
+    /// <summary>
+    /// Class representing arguments of on raid update event.
+    /// </summary>
+    public class OnRaidUpdateArgs : EventArgs
+    {
+        /// <summary>
+        /// The channel ID the event came from
+        /// </summary>
+        public int ChannelId;
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Events/OnRaidUpdateArgs.cs]: Same here (string as a channel/user id)

File: TwitchLib.PubSub/Events/OnRaidUpdateArgs.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        /// <summary>
+        /// Property representing the id of the raid event
+        /// </summary>
+        public Guid Id;
+        /// <summary>
+        /// Property representing the target channel id
+        /// </summary>
+        public int TargetChannelId;
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Events/OnRaidUpdateArgs.cs]: Same here (string as a channel/user id)

File: TwitchLib.PubSub/Events/OnRaidUpdatev2Args.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    /// <inheritdoc />
+    /// <summary>
+    /// Object representing the arguments for a raid go event
+    /// </summary>
+    public class OnRaidUpdateV2Args : EventArgs
+    {
+        /// <summary>
+        /// The channel ID the event came from
+        /// </summary>
+        public int ChannelId;
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Events/OnRaidUpdatev2Args.cs]: Same here (string as a channel/user id)

File: TwitchLib.PubSub/Events/OnRaidUpdatev2Args.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        /// <summary>
+        /// Property representing the id of the raid event
+        /// </summary>
+        public Guid Id;
+        /// <summary>
+        /// Property representing the target channel id
+        /// </summary>
+        public int TargetChannelId;
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Events/OnRaidUpdatev2Args.cs]: Same here (string as a channel/user id)

File: TwitchLib.PubSub/Models/Responses/Messages/RaidEvents.cs [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        /// <summary>
+        /// Raid id
+        /// </summary>
+        /// <value>The raid id</value>
+        public Guid Id { get; protected set; }
+        /// <summary>
+        /// Channel Id
+        /// </summary>
+        /// <value>The channel id</value>
+        public int ChannelId { get; protected set; }
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Models/Responses/Messages/RaidEvents.cs]: Same here (string as a channel/user id)

File: TwitchLib.PubSub/Models/Responses/Messages/RaidEvents.cs [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        /// <summary>
+        /// Target Channel Id
+        /// </summary>
+        /// <value>The target channel id</value>
+        public int TargetChannelId { get; protected set; }
```
Comment (swiftyspiffy) [inline TwitchLib.PubSub/Models/Responses/Messages/RaidEvents.cs]: Same here (string as a channel/user id)

**Comment by JayJay1989** [comment]
Also changed the Leaderboard event

**Comment by swiftyspiffy** [comment]
Looks good, thank you!

**Pull request merged**